doc/event-alarm-framework/event-alarm-framework.md

```diff
@@ -0,0 +1,1100 @@
+           +--rw warning?         uint64
+           +--rw informational?   uint64
+           +--rw acknowledged?    uint64
+
+  rpcs:
+    +---x acknowledge-alarms
+    |  +---w input
+    |  |  +---w id*   string
+    |  +--ro output
+    |     +--ro status?          int32
+    |     +--ro status-detail?   string
+    +---x unacknowledge-alarms
+    |  +---w input
+    |  |  +---w id*   string
+    |  +--ro output
+    |     +--ro status?          int32
+    |     +--ro status-detail?   string
+    +---x show-alarms
+       +---w input
+       |  +---w (option)?
+       |     +--:(time)
+       |     |  +---w time
+       |     |     +---w begin?   yang:date-and-time
+       |     |     +---w end?     yang:date-and-time
+       |     +--:(last-interval)
+       |     |  +---w period?     enumeration
+       |     +--:(severity)
+       |     |  +---w severity?   event:severity-type
+       |     +--:(identifier)
+       |        +---w id
+       |           +---w begin?   string
+       |           +---w end?     string
+       +--ro output
+          +--ro status?          int32
+          +--ro status-detail?   string
+          +--ro ALARM
+             +--ro ALARM_LIST* [id]
+                +--ro id              uint64
+                +--ro resource?       string
+                +--ro text?           string
+                +--ro time-created?   yang:date-and-time
+                +--ro type-id?        string
+                +--ro severity?       event:severity-type
+                +--ro acknowledged?   boolean
+```
+
+openconfig alarms yang is defined at [here](https://github.com/openconfig/public/blob/master/release/models/system/openconfig-alarms.yang)
+
+### 3.3.2 CLI
+#### 3.3.2.1 Exec Commands
+```
+sonic# alarm acknowledge <seq-id-of-raised-alarm>
+```
+An operator can acknolwedge a raised alarm. This indicates that the operator is aware of the fault condition and considers the condition not catastrophic. 
+Acknowledging an alarm updates system health parameter and thereby system LED by removing the particular alarm from status consideration.
+
+The alarm record in the ALARM table is marked with is_acknowledged field set to true.
+
+```
+sonic# event profile <profile-name>
+```
+The command takes name of specified file, validates it for its syntax and values; merges it with its internal static map of events *static_event_map*.
+
+#### 3.3.2.2 Configuration Commands
+```
+sonic(config)# logging server <ip> [log|event] 
+```
+Note: The 'logging server' command is an existing, already supported command. 
+It is only enhanced to take either 'log' or 'event' to indicate either native syslog messages or syslog messages corresponding to events alone are sent to the remote host.
+Support with VRF/source-interface and configuring remote-port are all backward comaptible and will be applicable to either 'log' or 'event' options.
+
+#### 3.3.2.3 Show Commands
+```
+sonic# show system health
+System Health: Normal, Warning or Error
```

Comment:
Do we have SONiC native CLI commands also in the management framework? 
Do you mean that the SONiC native CLI command will be extended to support the new state? if so, please provide an example of how it will be integrated into the existing command and if there is a changes on it.

Reply:
The command I specified here is mgmt-framework command. I wasnt aware that the native CLI has a system health command. Will look in to and update/enhance accordingly.